### DISCUSSION TITLE
Build CommonJS instead of ES6

### DISCUSSION THREAD

**Discussion by flogy** [ionic-team/capacitor#2807] [Ideas]
Hi all,

I am using Capacitor together with [Gatsby](https://github.com/gatsbyjs/gatsby). This works without any problems.

However, I have some trouble when using the [capacitor-fcm](https://github.com/stewwan/capacitor-fcm) and it seems that the module type of the build output is causing it, as it is ES6 and not CommonJS.

The plugin's source code looks like it was created with the [plugin-template](https://github.com/ionic-team/capacitor/blob/master/plugin-template/tsconfig.json) in this repository.

**So my questions are:**

1. What was the reason to decide to go with ES6 modules and not CommonJS? As far as I have seen most libraries seem to use CommonJS, probably to make it widely available.
2. Would it be a possibility to use CommonJS instead? I could create a PR to migrate the plugin template if this is an option.

Don't get me wrong. I am not very familiar with the JS standards and babel, etc. Maybe I just use it in a wrong way..?

Thanks in advance and keep up the great work!

**Comment by DimosthenisK** [comment]
I assume ES6 modules were chosen because capacitor is written in TS.

**Comment by flogy** [comment]
Thanks for your reply. I understand, but still it would be no problem to transpile TS to CommonJS and thus make it widely available without problems. Wouldn't this be a good idea?

**Comment by imhoffd** [comment]
My understanding is that:
1. ES6 modules are more appropriate for the web
1. Modern bundlers can handle input using both module systems
1. Capacitor leans _almost uncomfortably_ towards modern standards, so this decision makes sense

If there's a particular issue with the plugin output and the latest stable version of webpack, rollup, etc, please open an issue. Thanks!
